logger/regression.go

```diff
@@ -0,0 +1,93 @@
+package logger
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+)
+
+type (
+	// RegressionLogger logs data associated with unique ids, with optional
+	// meta data.
+	RegressionLogger interface {
+		Log(dataID string, data interface{}, meta map[string]interface{}) error
+	}
+
+	// RegressionLoggerCloser is a closable RegressionLogger, which is typically
+	// used when writing log to file.
+	RegressionLoggerCloser interface {
+		io.Closer
+		RegressionLogger
+	}
+
+	regressionLogger struct {
+		w io.WriteCloser
+	}
+
+	filterType string
+
+	// HeaderEntry contains a Key mapped to a regression log meta data Value.
+	HeaderEntry struct {
+		Key   string
+		Value string
+	}
+)
+
+var filters = marshalFilters(
+	"+qStateCounts",
+	"+qGrandTotalRow",
+	"+qDataPages",
+	"+qPivotDataPages",
+	"+qStackedDataPages",
+	"-qNum",
+)
+
+func marshalFilters(filters ...filterType) []byte {
+	rawJson, err := json.Marshal(filters)
+	if err != nil {
+		panic(err)
+	}
+	return rawJson
+}
+
+// NewRegressionLogger creates a new RegressionLoggerCloser with headerEntries
+// written in the header of the log.
+func NewRegressionLogger(w io.WriteCloser, headerEntries ...HeaderEntry) RegressionLoggerCloser {
+	fmt.Fprintf(w, "HEADER_KEY\tHEADER_VALUE\n")
+	fmt.Fprintf(w, "FILTERS\t%s\n", filters)
+	for _, he := range headerEntries {
+		fmt.Fprintf(w, "%s\t%s\n", strings.ToUpper(strings.TrimSpace(he.Key)), strings.TrimSpace(he.Value))
+	}
+	fmt.Fprintln(w, "---")
+	fmt.Fprintln(w, "ID\tMETA\tDATA")
```

Comment:
So we'll have a tsv with different amount of columns depending on row position? Is this format digestible by Sense etc?

Reply:
Well, if we could start reading from after `---`. I have not tried yet.

Reply:
you can set the load script in sense to skip the first X rows

Reply:
Okay. Then we would have to fix the number of rows in the first part, or chose number of rows dynamically searching for "---". 

My idea was for us to be able to add new metadata rows in the first part (like `ID_FORMAT`).

Reply:
yea fixed would be better from sense perspective